docs/docs/gatsby-core-philosophy.md

```diff
@@ -2,7 +2,177 @@
 title: Gatsby core philosophy
 ---
 
-This is a stub. Help our community expand it.
+Gatsby's core philosophy can be divided into three parts.
 
-Please use the [Gatsby Style Guide](/contributing/gatsby-style-guide/) to ensure your
-pull request gets accepted.
+**First, our vision.** Our vision is to _create better building blocks for the web_, and _build a cohesive "content mesh system"_, in order to _make building websites fun_.
+
+**Second, our tooling philosophy.** In order to make website building fun, Gatsby's tooling must embody certain qualities. These include:
+
+- making the right thing the easy thing
+- progressively disclosure of complexity
+
+**Third, our community philosophy.** We can't do this alone, so we're striving to build the most inclusive community on the web. For that reason, we _work in the open_. At all times, we believe, and strive to communicate, that _you belong here_.
+
+## The Gatsby Vision
+
+### Create new, higher-level web building blocks
+
+Today's building blocks for the web are HTML, CSS, and Javascript; `<img>` and `<a href>` and so on.
+
+We believe that in 20 or 25 years, we'll look back at many of these blocks like we look back at machine code or assembly language today; low-level languages that are great compile targets for higher-level languages that are easier to write in.
```

Comment:
I think we need to make extra clear that HTML and CSS demand respect because of their impact on users. They can be hand-coded (and likely still will), but perhaps we could say that sophisticated functionality and quality can be baked in to Gatsby tooling with little effort on the part of the website author?